### DISCUSSION TITLE
How to ensure second screen is focused after first one is blurred

### DISCUSSION THREAD

**Discussion by AlixH** [react-navigation/react-navigation#12323] [Q&A]
I have 2 screens in a Stack Navigator. Both use the same hook which sets the navbar color on focus, and reset it on blur. Thing is, when navigating from screen A to screen B, screen B sets the color and then screen A resets it. Conclusion, In the end, screen B does not have the right color. I have tried using useFocusEffect hook, without success so far. The effect of the second screen (B) still runs after the cleanup of the first screen. I have seen this issue mentioning the issue have been fixed in 2020, but still not working for me: https://github.com/react-navigation/react-navigation/issues/7963

**Comment by satya164** [comment]
> The effect of the second screen (B) still runs after the cleanup of the first screen

This sounds like the expected behavior and wouldn't cause "navigating from screen A to screen B, screen B sets the color and then screen A resets it" since second dcreens effect (set) runs after the first screen's cleanup (reset)

**Reply by AlixH** [reply]
Sorry it is a typo, it runs before

**Reply by satya164** [reply]
Can you share a repro with the issue?
